javaslang/src/main/java/javaslang/collection/Tree.java

```diff
@@ -748,12 +771,12 @@ public boolean isEmpty() {
 
         @Override
         public boolean isLeaf() {
-            return children.isEmpty();
+            return size == 1;
         }
 
         @Override
         public int size() {
-            return 1 + children.foldLeft(0, (acc, child) -> acc + child.length());
+            return size;
```

Comment:
@paplorinc Sorry for the late response - I returned from holidays.

I recently removed all lazy types from internal collection instance vars like `size` and `hash`. For now please do not use Lazy for size. We trade memory for cpu cycles. If needed we may change it later without the need to change public API or Tree semantics.